### PR TITLE
[1.5.x] BXMSPROD-1394 Quarkus to 1.11.7 from 1.11.6

### PR DESCRIPTION
**Pull request by Ginxo**
Many thanks for submitting your Pull Request :heart:! 
Ticket: https://issues.redhat.com/browse/BXMSPROD-1394

Please make sure that your PR meets the following requirements:

- [ ] You have read the [contributors guide](CONTRIBUTING.md)
- [ ] Your code is properly formatted according to [this configuration](https://github.com/kiegroup/kogito-runtimes/tree/master/kogito-build/kogito-ide-config)
- [ ] Pull Request title is properly formatted: `KOGITO-XYZ Subject`
- [ ] Pull Request title contains the target branch if not targeting master: `[0.9.x] KOGITO-XYZ Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains link to any dependent or related Pull Request
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>Run all builds</b>  
  Please add comment: <b>Jenkins retest this</b>

* <b>Run (or rerun) specific test(s)</b>  
  Please add comment: <b>Jenkins (re)run [runtimes|optaplanner|apps|examples] tests</b>
 
* <b>Quarkus LTS checks</b>  
  Please add comment: <b>Jenkins run LTS</b>

* <b>Run (or rerun) LTS specific test(s)</b>  
  Please add comment: <b>Jenkins (re)run [runtimes|optaplanner|apps|examples] LTS</b>

* <b>Native checks</b>  
  Please add comment: <b>Jenkins run native</b>

* <b>Run (or rerun) native specific test(s)</b>  
  Please add comment: <b>Jenkins (re)run [runtimes|optaplanner|apps|examples] native</b>

* <b>Full Kogito testing</b> (with cloud images and operator BDD testing)  
  Please add comment: <b>Jenkins run BDD</b>  
  <b>This check should be used only if a big change is done as it takes time to run, need resources and one full BDD tests check can be done at a time ...</b>
</details>

### CODE REVIEW

**Comment by Ginxo** [comment]
@danielezonca @cristianonicolai any idea about
`ArtifactNotFoundException: Could not find artifact org.kie.kogito:kogito-quarkus-rules-integration-test-legacy`
`ArtifactNotFoundException: Could not find artifact org.kie.kogito:kogito-quarkus-rules-integration-test-session`
`ArtifactNotFoundException: Could not find artifact org.kie.kogito:kogito-quarkus-decisions-integration-test`
`ArtifactNotFoundException: Could not find artifact org.kie.kogito:kogito-quarkus-predictions-integration-test`

**Comment by jstastny-cz** [comment]
The fails are all similar. Seems suspiciously like problem in quarkus artifact resolution :/ 
```
07:12:36,728 [ERROR] testAdult  Time elapsed: 0.014 s  <<< ERROR!
java.lang.RuntimeException: io.quarkus.bootstrap.BootstrapException: Failed to create the application model for org.kie.kogito:kogito-quarkus-decisions-integration-test::jar:1.5.1-SNAPSHOT
.
.
.
Caused by: org.eclipse.aether.transfer.ArtifactNotFoundException: Could not find artifact org.kie.kogito:kogito-quarkus-decisions-integration-test:jar:1.5.1-SNAPSHOT
        at org.eclipse.aether.internal.impl.DefaultArtifactResolver.resolve(DefaultArtifactResolver.java:414)
        ... 49 more

```

so it's failing at finding the artifact itself ...

**Comment by evacchi** [comment]
@aloubyansky are you aware of one such issue with 1.11.7 [1]?

@jstastny-cz @Ginxo I'd say if we are in a rush we have two choices (1) we ignore the failure and merge nonetheless (pretty risky, especially because I fear the release pipeline would break (2) we stay on 1.11.6 for our own build (1.11.7 should still work if used with our extension)

in the meantime I am trying this locally


[1] interestingly enough I am seeing a similar issue on Quarkus 2 on ecosystem ci https://github.com/kiegroup/kogito-runtimes/runs/2969254706?check_suite_focus=true

**Comment by Ginxo** [comment]
@evacchi many thanks man!! I usually choose the safest way in cases like this... So let's wait for your local testing

**Comment by aloubyansky** [comment]
It might also be related to the issue I was investigating last week. There is some serious issue with your CI but I'm not yet sure what the issue is.

**Comment by aloubyansky** [comment]
This specific one could be related to the way you clone repos. I guess you clone into `kogito-runtimes` and then you clone again under `kogito-runtimes`. So you end up with a nested repo. It may or may not be the cause.

**Comment by evacchi** [comment]
@aloubyansky I just tried this locally and I have no issue by first `mvn install -DskipTests` and then `mvn install` could it be this?

**Comment by aloubyansky** [comment]
@evacchi try the CI layout locally. I.e. do a nested clone. It does look like that's the cause. I think I already fixed in later versions.

**Comment by aloubyansky** [comment]
So, test the nested repo, not the "outer" one.

**Comment by evacchi** [comment]
@Ginxo I wonder if changing this line :

https://github.com/kiegroup/kogito-runtimes/blob/1.5.x/Jenkinsfile#L61

                    mvnCmd.run('clean install')

into

                    mvnCmd.run('clean install -DskipTests')
                    mvnCmd.run('install')


could fix the issue for now? 🤔  -- I would prefer not to fiddle too much with the dir layout on this branch @aloubyansky

**Comment by aloubyansky** [comment]
Sure, but it'd be good to know the cause anyway.

**Comment by evacchi** [comment]
yes sure I'll try it rn @aloubyansky

**Comment by aloubyansky** [comment]
Make sure you clean your local repo when you test that @evacchi

**Comment by evacchi** [comment]
> Make sure you clean your local repo when you test that @evacchi

reproduced. Yes the directory layout is the culprit. I am trying if just removing the `pom.xml` from the root directory could do the trick -- I don't think the outer project is being used at all, it may be even `rm -rf ./*` for that matter
another trick may be to clone the nested repos into a subfolder e.g. `K/kogito-runtimes`

**Comment by aloubyansky** [comment]
BTW, I think this particular issues has been fixed in later versions of Quarkus.

**Comment by aloubyansky** [comment]
I believe this was the fix https://github.com/quarkusio/quarkus/pull/16751

**Comment by evacchi** [comment]
I am running locally it seems to work so far, try this @Ginxo https://github.com/Ginxo/kogito-runtimes/pull/4

**Comment by Ginxo** [comment]
retrying https://eng-jenkins-csb-business-automation.apps.ocp4.prod.psi.redhat.com/job/KIE/job/kogito/job/pullrequest/job/kogito-runtimes/453/ with @evacchi's suggestion

**Comment by evacchi** [comment]
I don't think you have merged it @Ginxo

**Comment by Ginxo** [comment]
> I don't think you have merged it @Ginxo

No, i didn't, just replayed job adding your changes directly from jenkins (I will in case we agree this is something to commit)

**Comment by Ginxo** [comment]
Jenkins retest this

**Comment by radtriste** [comment]
Jenkins retest this

File: Jenkinsfile [hunk 1]
```diff
@@ -11,7 +11,7 @@ pipeline {
         label 'kie-rhel7 && kie-mem16g'
     }
     tools {
-        maven 'kie-maven-3.6.2'
+        maven 'kie-maven-3.6.3'
```
Comment (evacchi) [inline Jenkinsfile:14]: Are we ok with this @radtriste ? Looks ok to me, just fyi

**Review by evacchi — APPROVED** [review]
Looks fine just want to double check with @radtriste If upgrading maven to a minor is fine on this branch

File: Jenkinsfile (hunk 1 above)
Comment (radtriste) [reply Jenkinsfile:14]: should be ok. I am suprised it was still with `3.6.2` XD

File: Jenkinsfile [hunk 2]
```diff
@@ -34,6 +34,10 @@ pipeline {
                     checkoutRepo('kogito-examples')
                     checkoutRepo('kogito-examples', 'kogito-examples-persistence')
                     checkoutRepo('kogito-examples', 'kogito-examples-events')
+
+                    // remove pom.xml from parent overlapping workspace (it is not really used.)
+                    // Work-around an issue with Quarkus' resolver with our project layout
+                    sh 'rm ./pom.xml'
```
Comment (radtriste) [inline Jenkinsfile:40]: For this, we may use in the future the Jenkins job checkout to relative path option on all our jobs. Will open an issue for that

**Review by radtriste — APPROVED** [review]

**Comment by radtriste** [comment]
`Jenkins Tests / jenkinsfile-tests (pull_request)` failing is a problem with http usage in dependent pom.
This check can be ignored here as the Jenkinsfile change should not impact.

File: Jenkinsfile (hunk 2 above)
Comment (radtriste) [reply Jenkinsfile:40]: https://issues.redhat.com/browse/KOGITO-5475

**Pull request merged**